### DISCUSSION TITLE
Resize an RGB image

### DISCUSSION THREAD

**Discussion by vikashg** [Project-MONAI/MONAI#3689] [Q&A] [answered]
Hello I am trying to use a pipeline where I feed in RGB images. The images when read using `LoadImage()` gives an array of size (1024, 2678, 3). As mentioned in discussion #3635, the `Resize()` function assumes a channel first format. Following that, I tried 
```
transform_list = Compose([LoadImage(image_only=True), AsChannelFirst(), Resize((500, 500))])
x = transform_list('filename.png')
print(x[0].shape)
```
The output is (500, 500). I fear that I loose the color information in 3 channels. What will be an appropriate method to resample a RGB image. 
Thanks

**Comment by Nic-Ma** [comment, marked as answer]
Hi @vikashg , may I know why you use `x[0].shape` instead of `x.shape`?
`x` still contains 3 channels for RGB.

Thanks.

**Reply by vikashg** [reply]
Ok because I thought that the transformlist will output a list of images as default. but I might have messed up. While we are on the topic. Suppose I need to train something on a 3 channel image, in monai the data format is (num_batch_size, num_channel, sizex, sizey) ? Is that right ?

**Reply by Nic-Ma** [reply]
transforms work for single image, DataLoader generates a batch data for network.

Thanks.
